test/e2e/volume_provisioning.go

```diff
@@ -184,6 +186,70 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
 			Expect(err).To(HaveOccurred())
 			framework.Logf(err.Error())
 		})
+
+		It("should test that deleting a claim before the volume is provisioned deletes the volume. [Volume]", func() {
+			// This case tests for the regressions of a bug fixed by PR #21268
+			// REGRESSION: Deleting the PVC before the PV is provisioned can result in the PV
+			// not being deleted.
+			// NOTE:  Polls until no PVs are detected, times out at 5 minutes.
+
+			const raceAttempts int = 100
+			var suffix string = ns[len(ns)-5:] // match trailing random string of generated namespace
+			var regressedPVs []v1.PersistentVolume
+			var regressCount int
+
+			By("Creating a StorageClass")
+			class := newStorageClass("kubernetes.io/gce-pd", suffix)
+			class, err := c.Storage().StorageClasses().Create(class)
+			Expect(err).NotTo(HaveOccurred())
+			defer c.Storage().StorageClasses().Delete(class.Name, nil)
+			framework.Logf("Created sc %s", class.Name)
+
+			// To increase chance of detection, attempt multiple iterations
+			By(fmt.Sprintf("Creating and deleting PersistentVolumeClaims %d times", raceAttempts))
+			claim := newClaim(ns, suffix, false)
+			for i := 0; i < raceAttempts; i++ {
+				tmpClaim, err := c.Core().PersistentVolumeClaims(ns).Create(claim)
+				Expect(err).NotTo(HaveOccurred())
+				err = c.Core().PersistentVolumeClaims(ns).Delete(tmpClaim.Name, nil)
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			By(fmt.Sprintf("Checking for residual PersistentVolumes associated with StorageClass %s", class.Name))
+			// Every 10 seconds, check for PVs that have not been deleted.
+			err = wait.Poll(10*time.Second, 300*time.Second, func() (bool, error) {
+				regressedPVs = []v1.PersistentVolume{}
+				regressCount = 0
+
+				allPVs, err := c.Core().PersistentVolumes().List(metav1.ListOptions{})
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
Not sure you want an `Expect` in this poll func. I think it might be better to return `err` which will stop the poll loop.